skills/cypilot/scripts/cypilot/utils/context.py

```
@@ -190,33 +191,230 @@ def get_known_id_kinds(self) -> Set[str]:
+            for sc in self.sources.values():
+                if not sc.reachable or sc.meta is None:
+                    continue
+                for art, _sys in sc.meta.iter_all_artifacts():
+                    art_path = (sc.path / art.path).resolve()
+                    if art_path.exists():
+                        try:
+                            for h in scan_cpt_ids(art_path):
+                                if h.get("type") == "definition" and h.get("id"):
+                                    ids.add(str(h["id"]))
+                        except Exception:
+                            continue
```

Ruff (0.15.2)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)